### PR TITLE
Move analytics services to the back office

### PR DESCRIPTION
**Pull request by PaulDoyle-DEFRA**
- Move analytics presentation services to the back office
- Ensure start_form view is always logged
- Ensure duplicate consecutive page_views are logged once only
https://eaflood.atlassian.net/browse/RUBY-2799
https://eaflood.atlassian.net/browse/RUBY-2865
https://eaflood.atlassian.net/browse/RUBY-2867

### CODE REVIEW

File: app/services/waste_carriers_engine/analytics/user_journey_service.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def run(transient_registration:, current_user: nil)
         @current_user = current_user
         @journey = find_or_create_user_journey(journey_type, token)
 
-        PageView.create!(page: page, time: Time.zone.now, route: route, user_journey: journey)
+        journey.page_views.create!(page:, route:, time: Time.zone.now)
```
Comment (brujeo) [inline app/services/waste_carriers_engine/analytics/user_journey_service.rb:30]: Should it be @journey.page_views.create! instead? I see `@journey` variable being set, but not `journey`

File: app/services/waste_carriers_engine/analytics/user_journey_service.rb [hunk 2]
```diff
@@ -27,7 +27,7 @@ def run(transient_registration:, current_user: nil)
...
 
         if UserJourney::COMPLETION_PAGES.include?(page)
           journey.complete_journey(transient_registration)
```
Comment (brujeo) [inline app/services/waste_carriers_engine/analytics/user_journey_service.rb:33]: same here. Is `journey` variable set?

File: app/services/waste_carriers_engine/analytics/user_journey_service.rb (hunk 1 above)
Comment (PaulDoyle-DEFRA) [reply app/services/waste_carriers_engine/analytics/user_journey_service.rb:30]: There's an `attr_accessor :journey` so `journey` also works here.

File: app/services/waste_carriers_engine/analytics/user_journey_service.rb (hunk 2 above)
Comment (PaulDoyle-DEFRA) [reply app/services/waste_carriers_engine/analytics/user_journey_service.rb:33]: As above

File: app/services/waste_carriers_engine/analytics/user_journey_service.rb (hunk 1 above)
Comment (brujeo) [reply app/services/waste_carriers_engine/analytics/user_journey_service.rb:30]: ok

**Review by brujeo — APPROVED** [review]

**Review by jjromeo — DISMISSED** [review]

**Review by jjromeo — DISMISSED** [review]

**Pull request merged**